erpnext/hr/doctype/holiday_list/holiday_list.py

```diff
@@ -62,6 +63,53 @@ def get_weekly_off_date_list(self, start_date, end_date):
 	def clear_table(self):
 		self.set('holidays', [])
 
+	def on_update(self):
+		calendar_events = frappe.get_all("Event", filters={"holiday_list": self.name}, fields=["name", "starts_on"])
+		if not calendar_events:
+			for holiday in self.holidays:
+				if holiday.is_weekly_off:
+					continue
+
+				frappe.get_doc({
+					"doctype": "Event",
+					"subject": "Holiday " + holiday.holiday_date,
+					"starts_on": holiday.holiday_date,
+					"event_category": "Holiday",
+					"holiday_list": self.name,
+					"event_type": "Public",
+					"all_day":1
+				}).insert(ignore_permissions=True)
+
+		else:
+			holidays = []
+			for holiday in self.holidays:
+				if holiday.is_weekly_off:
+					continue
+				holidays.append(frappe.utils.get_datetime_str(holiday.holiday_date))
```

Comment:
```suggestion
			holidays = [frappe.utils.get_datetime_str(holiday.holiday_date) for holiday in self.holidays if holiday.is_weekly_off]
```

Reply:
fixed with below code
holidays = [frappe.utils.get_datetime_str(holiday.holiday_date) for holiday in self.holidays if not holiday.is_weekly_off]